### PR TITLE
chore: do not rely on Docker

### PR DESCRIPTION
**Pull request by cragwolfe**
Do not rely on Docker in the development environment, since it adds unnecessary overhead to run notebooks locally, etc.

Includes README update with clearer instructions on how to get the python development running.

Note, there are still some un-advertised (as in do not appear in output of `make`) docker- make targets for convenience. These are not needed for development, however.

**Testing**

Make sure `run-notebooks` works as expected, and one step through the [pipeline-notebooks](https://github.com/Unstructured-IO/pipeline-sec-filings/tree/main/pipeline-notebooks) and [exploration-notebooks](https://github.com/Unstructured-IO/pipeline-sec-filings/tree/main/exploration-notebooks).

`make` output is sane, with no reference to docker and new target for `run-jupyter`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,47 +57,52 @@ pip-compile:
...
-
-## run-app-dev:                 runs the FastAPI api with hot reloading
-.PHONY: run-app-dev
-run-app-dev:
+#########
+
+## run-jupyter:                 starts jupyter notebook
+.PHONY: run-jupyter
+run-jupyter:
+	PYTHONPATH=$(realpath .) JUPYTER_PATH=$(realpath .) jupyter-notebook  --no-browser --NotebookApp.token='' --NotebookApp.password=''
```
Comment (qued) [inline Makefile:75]: I think the most common use case would be that along with running the server, the user would also want a browser window open to the Jupyter file explorer. With that in mind could we drop the --no-browser flag?

File: README.md [hunk 2]
```diff
@@ -7,34 +7,29 @@
...
-options for installing and/or running the pipeline.
+## Developer Quick Start
 
-To install all of the dependencies for the pipeline, run `make install`. The pipeline is intended
-to be run from the base directory of this repo. If you want to run the pipeline from another
-directory, ensure the base directory of the repo is added to your Python path. You can do that
-by running `export PYTHONPATH=${PWD}:${PYTHONPATH}` from this directory.
+* Using `pyenv` to manage virtualenv's is recommended
+	* Mac install instructions:
+		*  `pyenv brew install pyenv-virtualenv`
```
Comment (qued) [inline README.md:18]: It looks like something is missing from this.

File: README.md [hunk 3]
```diff
@@ -143,7 +138,16 @@ and `open_form_by_ticker('mmm', '10-K', your_organization_name, your_email)` ope
...
+The python module [section.py](/blob/main/prepline_sec_filings/api/section.py) contains the FASTApi code needed to serve the API. It's created with `make generate-api`, which derives the API from the notebook [pipeline-section.ipynb](/blob/main/pipeline-notebooks/pipeline-section.ipynb). 
+
+You can generate the FastAPI APIs from all [pipeline-notebooks/](/pipeline-notebooks) by running `make generate-api`.  
+
+## Docker
+
+It is not necessary to run Docker in a local development environment, however a Dockerfile and
+make targets of `docker-build`, `docker-start-api`, and `docker-start-jupyter` are provided for convenience.
+
+You can also launch a Jupyter instance to try out the notebookes with [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/Unstructured-IO/pipeline-sec-filings/HEAD).
```
Comment (qued) [inline README.md:150]:
```suggestion
You can also launch a Jupyter instance to try out the notebooks with [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/Unstructured-IO/pipeline-sec-filings/HEAD).
```

**Review by qued — APPROVED** [review]
Other than dropping `--no-browser` from the target and documentation nits, LGTM!

File: README.md (hunk 2 above)
Comment (cragwolfe) [reply README.md:18]: Not sure what you mean. This is what the rendering looks like.

image

Comment (cragwolfe) [reply README.md:18]: Verbose dev instructions is a TODO for https://github.com/Unstructured-IO/community

Comment (qued) [reply README.md]: Sorry should have been more specific.

I'm pretty sure `pyenv brew install pyenv-virtualenv` won't work. It's probably meant to be `brew install pyenv pyenv-virtualenv` or broken up into 2 lines if `brew` doesn't support that syntax.

Comment (cragwolfe) [reply README.md]: fixed the brew command, though

**Pull request merged**